File: servers/elixir/lib/haveapi/validator/acceptance.ex
defmodule HaveAPI.Validator.Acceptance do
  use HaveAPI.Validator

  def name, do: :accept

  def init(opts) do
    %{
      value: Keyword.fetch!(opts, :value),
      message: opts[:message] || "has to be '#{inspect(opts[:value])}'"
    }
  end

  def validate(%{value: v}, v, _params), do: :ok
  def validate(opts, _v, _params), do: {:error, [opts.message]}
end

File: servers/ruby/lib/haveapi/validators/acceptance.rb
require 'haveapi/validator'

module HaveAPI
  # Accepts a single configured value.
  #
  # Short form:
  #   string :param, accept: 'value'
  #
  # Full form:
  #   string :param, accept: {
  #     value: 'value',
  #     message: 'the error message'
  #   }
  class Validators::Acceptance < Validator
    name :accept
    takes :accept

    def setup
      @value = if simple?
                 take

               else
                 take(:value)
               end

      @message = take(
        :message,
        HaveAPI.message('haveapi.validators.acceptance.accepted', value: @value)
      )
    end

    def describe
      {
        value: @value,
        message: HaveAPI.localize(@message)
      }
    end

    def valid?(v)
      v == @value
    end
  end
end
